src/server/replica.h

```diff
@@ -80,12 +84,28 @@ class Replica {
   void DefaultErrorHandler(const GenericError& err);
 
  private: /* Main dlfly flow mode functions */
+  struct MultiShardExecution {
+    boost::fibers::mutex map_mu;
+
+    struct TxExecutionSync {
+      boost::fibers::barrier berrier;
+      std::atomic_uint32_t counter;
+      TxExecutionSync(uint32_t counter) : berrier(counter), counter(counter) {
+      }
+    };
+
+    std::unordered_map<TxId, TxExecutionSync> tx_sync_execution;
```

Comment:
Lets not interleave members and struct declarations, so all fields at the bottom and all decls at the top - easier to read